proof/proof.go

```diff
@@ -301,11 +301,11 @@ type Proof struct {
 	// re-derivation of the asset group key.
 	GenesisReveal *asset.Genesis
 
-	// GroupKeyReveal is an optional set of bytes that represent the public
-	// key and Tapscript root used to derive the final tweaked group key for
-	// the asset group. This field must be provided for issuance proofs of
-	// grouped assets.
-	GroupKeyReveal *asset.GroupKeyReveal
+	// GroupKeyReveal contains the data required to derive the final tweaked
+	// group key for an asset group.
+	//
+	// NOTE: This field is mandatory for issuance proofs of grouped assets.
```

Comment:
IIRC this only needs to be set for the group anchor. Then each following tranche just needs to have a valid signature for the previously revealed group key. Just in case we want to add some extra precision to this sentence.